src/table/src/requests.rs

```diff
@@ -49,7 +55,83 @@ pub struct CreateTableRequest {
     pub region_numbers: Vec<u32>,
     pub primary_key_indices: Vec<usize>,
     pub create_if_not_exists: bool,
-    pub table_options: HashMap<String, String>,
+    pub table_options: TableOptions,
+}
+
+#[derive(Debug, Clone, Default, Serialize, Deserialize, PartialEq, Eq)]
+pub struct TableOptions {
+    /// Memtable size of memtable.
+    pub write_buffer_size: Option<ReadableSize>,
+    /// Time-to-live of table. Expired data will be automatically purged.
+    #[serde(with = "humantime_serde")]
+    pub ttl: Option<Duration>,
+    /// Extra options that may not applicable to all table engines.
+    pub extra_options: HashMap<String, String>,
+}
+
+pub const WRITE_BUFFER_SIZE_KEY: &str = "write_buffer_size";
+pub const TTL_KEY: &str = "ttl";
+
+impl TryFrom<&HashMap<String, String>> for TableOptions {
+    type Error = error::Error;
+
+    fn try_from(value: &HashMap<String, String>) -> Result<Self, Self::Error> {
+        let mut options = TableOptions::default();
+        if let Some(write_buffer_size) = value.get(WRITE_BUFFER_SIZE_KEY) {
+            let size = ReadableSize::from_str(write_buffer_size).map_err(|_| {
+                ParseTableOptionSnafu {
+                    key: WRITE_BUFFER_SIZE_KEY,
+                    value: write_buffer_size,
+                }
+                .build()
+            })?;
+            options.write_buffer_size = Some(size)
+        }
+
+        if let Some(ttl) = value.get(TTL_KEY) {
```

Comment:
We can use remove instead of get, then the rest is extra_options.

Reply:
> We can use remove instead of get, then the rest is extra_options.

The same as above, we cannot consume/mutate the hashmap in some cases.